### PR TITLE
Add hackeroneparcer 359

### PR DESCRIPTION
**Pull request by blacklotos**
**Note: DefectDojo is now on Python3 and Django 2.2.1 Please submit your pull requests to the 'dev' branch as the 'legacy-python2.7' branch is only for bug fixes. Any new features submitted to the legacy branch will be ignored and closed.**

When submitting a pull request, please make sure you have completed the following checklist:

- [x] Your code is flake8 compliant 
- [x] Your code is python 3.5 compliant
- [x] If this is a new feature and not a bug fix, you've included the proper documentation in the ReadTheDocs documentation folder. https://github.com/DefectDojo/Documentation/tree/master/docs or provide feature documentation in the PR.
- [ ] Model changes must include the necessary migrations in the dojo/dd_migrations folder.
- [x] Add applicable tests to the unit tests.

### CODE REVIEW

File: dojo/unittests/test_h1_parser.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+from dojo.models import Test
+
+
+class TestHackerOneParser(TestCase):
+
+    def test_parse_without_file_has_no_findings(self):
+        parser = HackerOneJSONParser(None, Test())
+        self.assertEqual(0, len(parser.items))
+
+    def test_parse_file_with_no_vuln_has_no_findings(self):
```
Comment (madchap) [inline dojo/unittests/test_h1_parser.py]: Nit-picking here, but if "no" finding, no 's' at findingS.

File: dojo/unittests/test_h1_parser.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        testfile = open("dojo/unittests/scans/h1/data_empty.json")
+        parser = HackerOneJSONParser(testfile, Test())
+        self.assertEqual(0, len(parser.items))
+
+    def test_parse_file_with_one_vuln_has_one_findings(self):
```
Comment (madchap) [inline dojo/unittests/test_h1_parser.py]: Same here.

File: dojo/unittests/test_h1_parser.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        testfile = open("dojo/unittests/scans/h1/data_one.json")
+        parser = HackerOneJSONParser(testfile, Test())
+        self.assertEqual(1, len(parser.items))
+
+    def test_parse_file_with_multiple_vuln_has_multiple_finding(self):
```
Comment (madchap) [inline dojo/unittests/test_h1_parser.py]: Nit-picking again, on the contrary, multiple findingS would take an s.

File: dojo/templates/dojo/import_scan_results.html [hunk 4]
```diff
@@ -52,6 +52,7 @@ <h3> Add Tests</h3>
 	<li><b>Fortify</b> - Import Findings from XML file format.</li>
 
 	<li><b>Generic Findings Import</b> - Import Generic findings in CSV format.</li>
+	<li><b>HackerOne Cases Import</b> - Import HackerOne cases findings in JSON format.</li>
```
Comment (madchap) [inline dojo/templates/dojo/import_scan_results.html]: Please re-order alphabetically.

File: dojo/tools/factory.py [hunk 5]
```diff
@@ -65,10 +65,12 @@
 from dojo.tools.hadolint.parser import HadolintParser
 from dojo.tools import SCAN_SONARQUBE_API
 from dojo.tools.aqua.parser import AquaJSONParser
+from dojo.tools.h1.parser import HackerOneJSONParser
 from dojo.tools.xanitizer.parser import XanitizerXMLParser
 from dojo.tools.trivy.parser import TrivyParser
 
 
+
```
Comment (madchap) [inline dojo/tools/factory.py]: Will not pass flake8 I would assume. Please make sure you run flake8 tests and fix warnings (and check the box in the PR description when done).

File: dojo/urls.py [hunk 6]
```diff
@@ -152,17 +150,7 @@
...
-        title="Defect Dojo API",
-        default_version='v2',
-        description="To use the API you need be authorized.",
-    ),
-    # if public=False, includes only endpoints the current user has access to
-    public=True,
-    # The API of a OpenSource project should be public accessible
-    permission_classes=[permissions.AllowAny],
-)
+schema_view = get_swagger_view(title='Defect Dojo API v2')
```
Comment (Maffooch) [inline dojo/urls.py]: You are undoing #1664 here. Please pull recent changes.

**Comment by madchap** [comment]
Addresses #1629

File: dojo/fixtures/test_type.json [hunk 7]
```diff
@@ -521,6 +521,13 @@
 		},
 		"model": "dojo.test_type",
 		"pk": 166
+	},
+	{
+		"fields": {
+			"name": "HackerOne"
```
[the remark this answers is not in the document]
Comment (blacklotos) [reply dojo/fixtures/test_type.json]: done

File: dojo/templates/dojo/import_scan_results.html [hunk 8]
```diff
@@ -53,6 +53,7 @@ <h3> Add Tests</h3>
 
 	<li><b>Generic Findings Import</b> - Import Generic findings in CSV format.</li>
 	<li><b>Gosec Scanner </b> - Import Gosec Scanner findings in JSON format.</li>
+	<li><b>HackerOne Cases Import</b> - Import HackerOne cases findings in JSON format.</li>
```
[the remark this answers is not in the document]
Comment (blacklotos) [reply dojo/templates/dojo/import_scan_results.html]: Done

File: dojo/urls.py (hunk 6 above)
Comment (madchap) [reply dojo/urls.py]: Fixed through later rebase. Marking as resolved.

**Comment by madchap** [comment]
Successfully tested locally. It would be nice to simply tick the boxes that apply in the description ;-)

**Pull request merged**